user-management/client/src/lib/data.ts

```diff
@@ -0,0 +1,69 @@
+export const tokenKey = 'um.token';
+export function saveToken(token: string | undefined): void {
+  if (token) {
+    sessionStorage.setItem(tokenKey, token);
+  } else {
+    sessionStorage.removeItem(tokenKey);
+  }
+}
+
+export function readToken(): string {
+  const token = sessionStorage.getItem(tokenKey);
+  if (!token) throw new Error('No token found');
+  return token;
+}
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export async function readTodos(): Promise<Todo[]> {
+  const res = await fetch('/api/todos', {
+    headers: {
+      Authorization: `Bearer ${readToken()}`,
+    },
+  });
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function insertTodo(todo: UnsavedTodo): Promise<Todo> {
+  const req = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${readToken()}`,
+    },
+    body: JSON.stringify(todo),
+  };
+  const res = await fetch('/api/todos', req);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function updateTodo(todo: Todo): Promise<Todo> {
+  const req = {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${readToken()}`,
+    },
+    body: JSON.stringify(todo),
+  };
+  const res = await fetch(`/api/todos/${todo.todoId}`, req);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function removeTodo(todoId: number): Promise<void> {
+  const req = {
+    method: 'DELETE',
+    Authorization: `Bearer ${readToken()}`,
```

Comment:
This one isn't actually used by the UI so you wouldn't notice the bug, but it is missing the "headers" field.